### PR TITLE
WIP - Adding to the test crd flows

### PR DESCRIPTION
**Pull request by kerencarmelwedel**
crd function to check noobaa crd commands added to the test

### CODE REVIEW

File: test/cli/test_cli_functions.sh [hunk 1]
```diff
@@ -284,10 +284,89 @@ function obc_cycle {
...
+    echo_time "checking if crd noobaa create command equals to crd status command"
+    for crd_status_array in ${crd_array[@]}
+    do
+        if [[ ${crd_create_array[@]} =~ ${crd_status_array} ]]
+        then            
+            echo_time "✅ ${crd_status_array} exists in the crds list"
+        else
+            echo_time "❌ ${crd_status_array} is not existed in the crds list"         
+        fi               
+    done
```
Comment (liranmauda) [inline test/cli/test_cli_functions.sh:303]: If the crd is not exists, why not exit?

File: test/cli/test_cli_functions.sh [hunk 2]
```diff
@@ -284,10 +284,89 @@ function obc_cycle {
...
 
+function crd_cycle {     
+    echo_time "Starting the crd cycle" 
+    local crd_create_array=($(test_noobaa silence crd create &>/dev/stdout | awk '{print $6}' | sed -e 's/[""\\]//g' | sed 's/.$//'))
+    local crd_status_array=($(test_noobaa silence crd status &>/dev/stdout | awk '{print $6}' | grep -v "Exists" | sed -e 's/[""\\]//g' | sed 's/.$//'))
+    crd_array=($(kubectl get crd | awk '{print $1}' | grep -v "NAME"))
+    
+    #comparing crd status and create arrays
+    echo_time "checking if crd noobaa create command equals to crd status command"
+    for crd_status_array in ${crd_array[@]}
```
Comment (liranmauda) [inline test/cli/test_cli_functions.sh:295]: This is not doing what the `echo_time` describes. 
it is checking that the crd status command is equal to the kubectl get crd command output.

File: test/cli/test_cli_functions.sh [hunk 3]
```diff
@@ -284,10 +284,89 @@ function obc_cycle {
...
+    done
+         
+    kuberun get crd  
+    if [ $? -ne 0 ]
+    then
+        echo_time "❌ kubectl get crd failed"
+        exit 1 
+    else 
+        echo_time "✅ kubectl get crd passed"    
+    fi
```
Comment (liranmauda) [inline test/cli/test_cli_functions.sh:312]: This is checking the kubrun function exit code.
why do we need to check kubrun/kubectl?

File: test/cli/test_cli_functions.sh [hunk 4]
```diff
@@ -284,10 +284,89 @@ function obc_cycle {
...
+        fi
+    done
+    kuberun get crd  
+    if [ $? -ne 0 ]
+    then
+        echo_time "❌ kubectl get crd failed"
+        exit 1 
+    else 
+        echo_time "✅ kubectl get crd passed"    
+    fi
```
Comment (liranmauda) [inline test/cli/test_cli_functions.sh:334]: This is checking the kubrun function exit code.
why do we need to check kubrun/kubectl?

File: test/cli/test_cli_functions.sh [hunk 5]
```diff
@@ -284,10 +284,89 @@ function obc_cycle {
...
+        exit 1 
+    else 
+        echo_time "✅ kubectl get crd passed"    
+    fi 
+
+    echo_time "creating crd again for checks..."     
+    test_noobaa timeout crd create 
+    local crd_after_create=($(test_noobaa silence crd status &>/dev/stdout | awk '{print $6}' | sed -e 's/[""\\]//g' | sed 's/.$//'))
+    #comparing crds after running create command
+    for crd_after_create in ${crd_array[@]}
```
Comment (liranmauda) [inline test/cli/test_cli_functions.sh:340]: it runs on the crd that we get using kubectl get crd command. it includes non-noobaa crds.

File: test/cli/test_cli_functions.sh [hunk 6]
```diff
@@ -284,10 +284,89 @@ function obc_cycle {
...
+    done
+
+    kuberun get crd  
+    if [ $? -ne 0 ]
+    then
+        echo_time "❌ kubectl get crd failed"
+        exit 1 
+    else 
+        echo_time "✅ kubectl get crd passed"    
+    fi
```
Comment (liranmauda) [inline test/cli/test_cli_functions.sh:357]: Same as above

File: test/cli/test_cli_functions.sh [hunk 7]
```diff
@@ -284,10 +284,89 @@ function obc_cycle {
...
+       
+    echo_time "✅  crd cycle is done"
+}
+
 function delete_backingstore_path {
     local object_bucket backing_store
     local backingstore=($(test_noobaa silence backingstore list | grep -v "NAME" | awk '{print $1}'))
-    local bucketclass=($(test_noobaa silence bucketclass list  | grep ${backingstore[1]} | awk '{print $1}'))
+    local bucketclass=($(test_noobaa bucketclass list  | grep ${backingstore[1]} | awk '{print $1}'))
+    echo "${bucketclass}"
```
Comment (liranmauda) [inline test/cli/test_cli_functions.sh:368]: Why are we printing the bucketclass here?

File: test/cli/test_cli_flow.sh [hunk 8]
```diff
@@ -64,6 +64,7 @@ function clean {
 }
 
 function main {
+    crd_cycle
```
Comment (liranmauda) [inline test/cli/test_cli_flow.sh:67]: Why crd cycle is here before the install?
How should we check the crd cycle if there is no NooBaa?

Comment (kerencarmelwedel) [reply test/cli/test_cli_flow.sh:67]: I consulted with Ohad and we thought its the best way, after noobaa is created it tries to delete crd while its not empty

File: test/cli/test_cli_functions.sh (hunk 7 above)
Comment (kerencarmelwedel) [reply test/cli/test_cli_functions.sh:368]: I will fix it for the next pr. Tried to check something here and forgot to delete it

File: test/cli/test_cli_functions.sh (hunk 5 above)
Comment (kerencarmelwedel) [reply test/cli/test_cli_functions.sh:340]: But they have exactly the same values...

File: test/cli/test_cli_functions.sh (hunk 1 above)
Comment (kerencarmelwedel) [reply test/cli/test_cli_functions.sh:303]: i'll add exit.

File: test/cli/test_cli_functions.sh (hunk 7 above)
Comment (liranmauda) [reply test/cli/test_cli_functions.sh:368]: Next commit 🙂

File: test/cli/test_cli_functions.sh (hunk 5 above)
Comment (liranmauda) [reply test/cli/test_cli_functions.sh:340]: They have also non-noobaa crds

File: test/cli/test_cli_flow.sh (hunk 8 above)
Comment (liranmauda) [reply test/cli/test_cli_flow.sh:67]: we can run this cycle prior to the install but also need to do it after the install and exclude the parts that we don't want to run.

File: test/cli/test_cli_functions.sh (hunk 3 above)
Comment (kerencarmelwedel) [reply test/cli/test_cli_functions.sh:312]: Asking again to make sure - you told me to run kubectl after every noobaa command i'm checking. Should i just run it as it is or should i check its' exit status? Or is just comparing between noobaa crd and kubectl get crd enough ?

Comment (liranmauda) [reply test/cli/test_cli_functions.sh:312]: We need to check that the kubectl returns all the expected crds.
so, we should have a list of crd (that is only noobaa’s), and then check that all appear in the kubectl get crd.
Cacking the exit code of our function does not do that.

File: test/cli/test_cli_functions.sh (hunk 5 above)
Comment (kerencarmelwedel) [reply test/cli/test_cli_functions.sh:340]: They don't, they have exactly the same values, at least how i wrote it

Comment (liranmauda) [reply test/cli/test_cli_functions.sh:340]: On minikube yes, on openshit there are more crds.

**Pull request closed without merging**